lib/misc/api/api_user.dart

```diff
@@ -42,3 +43,96 @@ Future<List<BaseItem>> fetchLocations() async {
 
   return location;
 }
+
+/// Get current user information
+Future getCurrentUser() async {
+  Map<String, dynamic> user = {};
+  final token = await api_token.getToken();
+
+  try {
+    await http.get(
+      Uri.parse('${api.getApiBaseUrl()}/user'),
+      headers: <String, String>{
+        'Content-Type': 'application/json; charset=UTF-8',
+        'Accept': 'application/json',
+        'Authorization': 'Bearer $token',
+      },
+    ).then((response) {
+      if (response.statusCode == 200) {
+        final data = jsonDecode(response.body);
+        user = {
+          'id': data['id'],
+          'first_name': data['first_name'],
+          'last_name': data['last_name'],
+          'email': data['email'],
+          'phone': data['phone_number'],
+          'location': data['location_id'],
+        };
+      } else {
+        if (kDebugMode) {
+          debugPrint('Request failed with status: ${response.statusCode}');
+        }
```

Comment:
Should properly return a message or something the like

Reply:
done